include/TextureManager.h

```diff
@@ -15,13 +32,59 @@ class TextureManager
 			for (auto& it : this->texture_cache) {
 				SDL_DestroyTexture(it.second);
 			}
+			for (auto& it : this->mapTile_texture_cache) {
+				SDL_DestroyTexture(it.second);
+			}
 		}
 
-		std::map<std::string, SDL_Texture*> texture_cache;
+		/*!
+		 * \brief Adds a single texture to the cache.
+		 * \param texture The texture identifier.
+		 * \param filePath The file path to the texture file.
+		 *
+		 * This function loads the texture from the specified file and stores it in
+		 * the `texture_cache`. If loading the texture fails, an exception is thrown.
+		 */
+		void addSingleTexture(Textures texture, const char* filePath);
 
-		SDL_Texture* loadTexture(const char* fileName);
+		/*!
+		 * \brief Adds multiple textures to the cache.
+		 * \param textures A map of texture identifiers and corresponding file paths.
+		 *
+		 * This function iterates over the provided map of textures and loads each
+		 * texture using `addSingleTexture`. It allows developers to add several
+		 * textures at once.
+		 */
+		void addTextures(const std::map<Textures, const char*>& textures);
+
+		/*!
+		 * \brief Loads a texture from the cache.
+		 * \param texture The texture identifier.
+		 * \return A pointer to the `SDL_Texture` if found, or `nullptr` if not found.
+		 *
+		 * This function looks up a texture in the `texture_cache` and returns the
+		 * corresponding `SDL_Texture*`. If the texture is not found, it logs an error
+		 * message and returns `nullptr`.
+		 */
+		SDL_Texture* loadTexture(Textures texture);
         static std::vector<SDL_Rect> splitSpriteSheet(SDL_Texture* spriteSheet, int width, int height, int spritesOnSheet);
 		static void draw(SDL_Renderer* renderer, SDL_Texture* texture, SDL_Rect src, SDL_Rect dest, bool flipped = false);
+
+		/*!
+		 * \brief Loads a map tile texture from the file system and caches it.
+		 * \param path The file path to the texture.
+		 * \return A pointer to the `SDL_Texture` representing the map tile.
+		 *
+		 * This function checks if the map tile texture is already cached. If not, it
```

Comment:
Caching does not concern the developer, it can be mentioned at the documentation for the class itself, but does not add any information to the usage of this function